src/execution_plan/execution_plan.c

```diff
@@ -321,135 +336,150 @@ static AR_ExpNode** _OrderClause_GetExpressions(const AST *ast) {
 	return exps;
 }
 
+/* Keep track after resolved variabels.
+ * add variabels modified/set by op to resolved. */
+static void _UpdateResolvedVariabels(rax *resolved, OpBase *op) {
+    assert(resolved && op);
+    if(!op->modifies) return;
+
+    int count = Vector_Size(op->modifies);
+    for(int i = 0; i < count; i++) {
+        char *var;
+        Vector_Get(op->modifies, i, &var);
+        raxInsert(resolved, (unsigned char*)var, strlen(var), NULL, NULL);
+    }
+}
+
 ExecutionPlan* _NewExecutionPlan(RedisModuleCtx *ctx, AST *ast, ResultSet *result_set) {
-    GraphContext *gc = GraphContext_GetFromTLS();
-    Graph *g = gc->g;
-    ExecutionPlan *execution_plan = (ExecutionPlan*)calloc(1, sizeof(ExecutionPlan));    
-    execution_plan->result_set = result_set;
-    Vector *ops = NewVector(OpBase*, 1);
+    Graph *g;
     OpBase *op;
+    Vector *ops;
+    rax *resolved;
+    QueryGraph *q;
+    GraphContext *gc;
+    size_t node_count;
+    size_t edge_count;
+    FT_FilterNode *filter_tree;
+    ExecutionPlan *execution_plan;
+
+    resolved = raxNew();
+    ops = NewVector(OpBase*, 1);
+    gc = GraphContext_GetFromTLS();
+    g = gc->g;
+    execution_plan = (ExecutionPlan*)calloc(1, sizeof(ExecutionPlan));
+    execution_plan->result_set = result_set;
```

Comment:
Is this initialization-before-assignment design something you want to adopt in general? I don't see the benefit, and it adds so many lines.

Reply:
I'm not sure, I feel like it's nice to have most if not all variables declared at the top, hopefully accommodated with a comment explaining the purpose of each, as you might have noticed already I find it hard to keep a single style.

Reply:
Declaring variables at the top feels like a good design to me, but I'd prefer to do things like
`GraphContext *gc = GraphContext_GetFromTLS();`
when possible (which it is for most of these variables).

I don't mind experimenting with styles, though, just wanted to weigh in!